sklearn/linear_model/cd_fast.pyx

```diff
@@ -363,7 +418,7 @@ def sparse_enet_coordinate_descent(double[:] w,
             startptr = endptr
 
         # tol *= np.dot(y, y)
-        tol *= ddot(n_samples, <DOUBLE*>&y[0], 1, <DOUBLE*>&y[0], 1)
+        tol *= dot(n_samples, <floating*>&y[0], 1, <floating*>&y[0], 1)
```

Comment:
It's not clear to me that these typecasts are still necessary. Have you tried removing them?